allennlp/data/dataset_readers/universal_dependencies_multilang.py

```diff
@@ -0,0 +1,171 @@
+from typing import Dict, Tuple, List
+import logging
+import itertools
+from collections import defaultdict
+import numpy as np
+
+from overrides import overrides
+
+from allennlp.data.dataset_readers.dataset_reader import DatasetReader
+from allennlp.data.fields import Field, TextField, SequenceLabelField, MetadataField
+from allennlp.data.instance import Instance
+from allennlp.data.token_indexers import SingleIdTokenIndexer, TokenIndexer
+from allennlp.data.tokenizers import Token
+from allennlp.data.dataset_readers.universal_dependencies import lazy_parse
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+def parse_file_paths(file_paths: Dict[str, str]):
+    """
+    Converts from allennlp.params.Params to a python dict.
+
+    Parameters
+    ----------
+    file_paths :  ``Dict[str, str]``, required.
+        The dictionary of identifier (e.g. "en" for English) to file path.
+
+    Returns
+    -------
+    A dictionary of identifier ("en","it" etc.) to file path.
+    """
+    return dict(file_paths)
+
+
+@DatasetReader.register("universal_dependencies_multilang")
+class UniversalDependenciesMultiLangDatasetReader(DatasetReader):
+    """
+    Wraps UniversalDependenciesDatasetReader to support multiple input files.
+    Reads consecutive cases from each input files by the given batch_size.
+
+    Notice: when using the alternate option, one should also use the ``instances_per_epoch``
+    option for the iterator. Otherwise, each epoch will loop infinitely.
```

Comment:
Each file can have a different amount of data (which is usually what happens in practice...). That's why it is better to just continuously alternate between the files to make sure that the training is balanced across sources. (Otherwise, one could just concatenate all files into one long file with some language identifier per sample).

Reply:
Ah, I see, that's a good point.  That's a great thing to include in the docstring.